src/vs/workbench/services/configuration/browser/configurationService.ts

```diff
@@ -77,7 +77,7 @@ export class WorkspaceService extends Disposable implements IWorkbenchConfigurat
 	private readonly localUserConfiguration: UserConfiguration;
 	private readonly remoteUserConfiguration: RemoteUserConfiguration | null = null;
 	private readonly workspaceConfiguration: WorkspaceConfiguration;
-	private cachedFolderConfigs: ResourceMap<FolderConfiguration>;
+	private cachedFolderConfigs: DisposableMap<URI, FolderConfiguration> = this._register(new DisposableMap());
```

Comment:
I do not think ResourceMap<T> is same as Map<Uri, T>. Therefore we cannot use DisposableMap. Can you please dispose manually from the map instead of using DisposableMap which disposes automatically?